apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConversionOptions.cs

```diff
@@ -46,6 +57,6 @@ private SpannerConversionOptions(bool useDBNull)
         /// Determines the right conversion options to use based on the connection string of the given connection string builder.
         /// </summary>
         internal static SpannerConversionOptions ForConnectionStringBuilder(SpannerConnectionStringBuilder builder) =>
-            builder == null ? Default : builder.UseClrDefaultForNull ? s_useClrDefaultForNull : s_useDBNullForNull;
+            builder == null ? Default : new SpannerConversionOptions(!builder.UseClrDefaultForNull, builder.UseSpannerNumericForDecimal, builder.UsePgNumericForDecimal);
```

Comment:
Depending on how complex this gets (with multiple options) we *may* want to have a set of "common configurations" and reuse the same objects.